tt_metal/tt_metal.cpp

```diff
@@ -124,18 +124,15 @@ namespace detail {
         #ifdef DEBUG_PRINT_SHARD
             std::cout << "Writing to Device Sharded " << std::endl;
         #endif
-        auto cores = buffer.all_cores();
-        auto core_bank_ids = buffer.core_bank_indices();
         auto core_host_page_ids = buffer.core_host_page_indices();
 
 
         int dev_page_index = 0;
-        for(int core_index = 0; core_index < cores.size(); core_index++){
-            auto core = cores[core_index];
-            auto bank_id = core_bank_ids[core_index];
+        for(int core_index = 0; core_index < buffer.num_cores(); core_index++){
```

Comment:
not related to this specific line but more of this function:

I think we should change this function to:

void WriteToDevice( const Shared)